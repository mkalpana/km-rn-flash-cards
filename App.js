import React from 'react';
import { StyleSheet, View, StatusBar } from 'react-native';
import { clearItems } from './utils/storage';
import { Provider } from 'react-redux';
import store from './utils/reduxStore';
import { setLocalNotification } from './utils/notification';
import { Tabs } from './utils/navigation';


export default class App extends React.Component {
  componentDidMount() {
    // clearItems();
    setLocalNotification();
  }
  render() {
    return (
      <Provider store={store}>
        <View style={styles.container}>
          <StatusBar hidden={false} />
          <Tabs />
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'flex-start',
    marginTop: 20
  },
});
