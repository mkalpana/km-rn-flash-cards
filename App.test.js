import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

describe('App', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<App/>).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
