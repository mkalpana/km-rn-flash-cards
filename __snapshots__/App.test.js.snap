// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "marginTop": 20,
    }
  }
>
  <View
    loaded={
      Array [
        0,
      ]
    }
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={undefined}
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "borderTopColor": "rgba(0, 0, 0, .3)",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "height": 49,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
          testID={undefined}
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              />
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(52, 120, 246, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "marginLeft": 0,
                "marginTop": 0,
                "textAlign": "center",
              }
            }
          >
            DECKS
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
          testID={undefined}
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              />
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(146, 146, 146, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "marginLeft": 0,
                "marginTop": 0,
                "textAlign": "center",
              }
            }
          >
            NEW DECK
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
